docs/api/index.md

```diff
@@ -2592,23 +2592,114 @@ Vitest provides utility functions to help you out through it's **vi** helper. Yo
-  - If mocking a module with a default export, you'll need to provide a `default` key within the returned factory function object. This is an ES modules specific caveat, therefore `jest` documentation may differ as `jest` uses commonJS modules. *Example:*
+  ::: warning
+  `vi.mock` works only for modules that were imported with `import` keyword. It doesn't work with `require`.
+
+  Vitest statically analyzes your files to hoist `vi.mock`. It means that you cannot use `vi` that was not imported directly from `vitest` package (for example, from some utility file). To fix this, always use `vi.mock` with `vi` imported from `vitest`, or enable [`globals`](/config/#globals) config option.
+  :::
+
+  If `factory` is defined, all imports will return its result. Vitest calls factory only once and caches result for all subsequent imports, until [`vi.unmock`](#vi-unmock) or [`vi.doUnmock`](#vi-dounmock) is called.
+
+  Unlike in `jest`, the factory can be asynchronous, so you can use [`vi.importActual`](#vi-importactual) inside to get the original module.
+
+  ::: warning
+  `vi.mock` is hoisted (in other words, _moved_) to **top of the file**. It means that whenever you wrote it (be it inside `beforeEach` or `test`), it will actually be called before that.
+
+  This also means that you cannot use any variables inside the factory that are defined outside of the factory.
+
+  If you need to use variables inside the factory, try [`vi.doMock`](#vi-domock). It works the same way, but isn't hoisted. Beware that it only mocks subsequent imports.
+  :::
+
+  ::: warning
+  If you are mocking a module with default export, you will need to provide a `default` key within the returned factory function object. This is an ES modules specific caveat, therefore `jest` documentation may differ as `jest` uses CommonJS modules. For example,
 
   ```ts
-  vi.mock('path', () => {
+  vi.mock('./path/to/module.js', () => {
     return {
       default: { myDefaultKey: vi.fn() },
       namedExport: vi.fn(),
       // etc...
     }
   })
   ```
+  :::
+
+  If there is a `__mocks__` folder alongside a file that you are mocking, and the factory is not provided, Vitest will try to find a file with the same name in `__mocks__` subfolder and use it as an actual module. If you are mocking a dependency, Vitest will try to find a `__mocks__` folder in the [root](/config/#root) of the project (default is `process.cwd()`).
+
+  For example, you have this file structure:
+
+  ```
+  - __mocks__
+    - axios.ts
+  - src
+    __mocks__
+      - increment.ts
+    - increment.ts
+  - tests
+    - increment.test.ts
+  ```
+
+  If you call `vi.mock` in a test file without a factory provided, it will find file in `__mocks__` folder to use as a module:
```

Comment:
```suggestion
  If you call `vi.mock` in a test file without a factory provided, it will find a file in the `__mocks__` folder to use as a module:
```